### PR TITLE
Update course details cell layout for large dynamic types

### PR DESCRIPTION
**Pull request by vargaat**
refs: MBL-16574
affects: Student, Teacher
release note: Fixed course details list layout on large accessibility sizes.

test plan:
- Set text accessibility size to the largest possible value.
- Go to course details.
- Cell contents shouldn't overlap.
- Cell icons should also scale with the text.

## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [x] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [x] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Review by idevainstr — APPROVED** [review]
QA +1

File: Core/Core/SwiftUIViews/DisclosureIndicator.swift [hunk 1]
```diff
@@ -29,12 +29,14 @@ public struct DisclosureIndicator: View {
 }
 
 public struct InstDisclosureIndicator: View {
+    @ScaledMetric private var uiScale: CGFloat = 1
```
Comment (szabinst) [inline Core/Core/SwiftUIViews/DisclosureIndicator.swift:32]: 👍 nice

**Review by szabinst — APPROVED** [review]
QA + 1

**Pull request merged**